### PR TITLE
Add tilde style code fence support

### PR DESCRIPTION
**Pull request by Waqar144**

### CODE REVIEW

File: markdownhighlighter.cpp [hunk 1]
```diff
@@ -811,86 +819,121 @@ void MarkdownHighlighter::highlightSyntax(const QString &text)
...
-        case HighlighterState::CodeINI:
-            iniHighlighter(text);
-            return;
-        case HighlighterState::CodeTaggerScript:
-             taggerScriptHighlighter(text);
-             return;
-        case HighlighterState::CodeVex:
-            loadVEXData(types, keywords, builtin, literals, others);
-            break;
+    case HighlighterState::CodeCpp:
```
Comment (pbek) [inline markdownhighlighter.cpp]: Is there a reason you changed the switch indent?

Comment (Waqar144) [reply markdownhighlighter.cpp]: That was Qt Creator. It just changes the indent everytime i make a change, and this one involved a lot of changes so everything in the switch is one \t back

Comment (pbek) [reply markdownhighlighter.cpp]: It looks uuuuugly and so far I used the previous styling everywhere.

Comment (Waqar144) [reply markdownhighlighter.cpp]: Yes I know. Reverted.
I just don't know why the editors always revert it. :laughing:

Comment (pbek) [reply markdownhighlighter.cpp]: Thank you :grin:

**Comment by pbek** [comment]
This PR doesn't seem to work with QON, see https://github.com/pbek/QOwnNotes/pull/1584

**Pull request merged**